lib/charms/mongodb/v0/mongodb_backups.py

```diff
@@ -0,0 +1,457 @@
+    wait_fixed,
+)
+
+# The unique Charmhub library identifier, never change it
+LIBID = "9f2b91c6128d48d6ba22724bf365da3b"
+
+# Increment this major API version when introducing breaking changes
+LIBAPI = 0
+
+# Increment this PATCH version before using `charmcraft publish-lib` or reset
+# to 0 if you are raising the major API version
+LIBPATCH = 1
+
+logger = logging.getLogger(__name__)
+
+S3_PBM_OPTION_MAP = {
+    "region": "storage.s3.region",
+    "bucket": "storage.s3.bucket",
+    "path": "storage.s3.prefix",
+    "access-key": "storage.s3.credentials.access-key-id",
+    "secret-key": "storage.s3.credentials.secret-access-key",
+    "endpoint": "storage.s3.endpointUrl",
+    "storage-class": "storage.s3.storageClass",
+}
+S3_RELATION = "s3-credentials"
+REMAPPING_PATTERN = r"\ABackup doesn't match current cluster topology - it has different replica set names. Extra shards in the backup will cause this, for a simple example. The extra/unknown replica set names found in the backup are: ([^,\s]+)([.] Backup has no data for the config server or sole replicaset)?\Z"
+PBM_CONFIG_FILE_PATH_K8S = "/etc/pbm_config.yaml"
+PBM_PATH = "/usr/bin/pbm"
+PBM_STATUS_CMD = [PBM_PATH, "status", "-o", "json"]
+MONGODB_SNAP_DATA_DIR = "/var/snap/charmed-mongodb/current"
+
+
+class ResyncError(Exception):
+    """Raised when pbm is resyncing configurations and is not ready to be used."""
+
+
+class SetPBMConfigError(Exception):
+    """Raised when pbm cannot configure a given option."""
+
+
+class PBMBusyError(Exception):
+    """Raised when PBM is busy and cannot run another operation."""
+
+
+class MongoDBBackups(Object):
+    """Manages MongoDB backups."""
+
+    def __init__(self, charm, substrate):
+        """Manager of MongoDB client relations."""
+        super().__init__(charm, "client-relations")
+        self.charm = charm
+        self._substrate = substrate
+
+        # s3 relation handles the config options for s3 backups
+        self.s3_client = S3Requirer(self.charm, S3_RELATION)
+        self.framework.observe(
+            self.s3_client.on.credentials_changed, self._on_s3_credential_changed
+        )
+        self.framework.observe(self.charm.on.create_backup_action, self._on_create_backup_action)
+        self.framework.observe(self.charm.on.list_backups_action, self._on_list_backups_action)
+        self.framework.observe(self.charm.on.restore_action, self._on_restore_action)
+
+    # TODO a better way for mocking this
+    def substrate(self) -> str:
+        """Returns the substrate of the charm."""
+        return self._substrate
+
+    def _on_s3_credential_changed(self, event: CredentialsChangedEvent):
+        """Sets pbm credentials, resyncs if necessary and reports config errors."""
+        # handling PBM configurations requires that MongoDB is running and the pbm snap is
+        # installed.
+        if not self.charm.db_initialised:
+            logger.error("Cannot set PBM configurations, MongoDB has not yet started.")
```

Comment:
I think it is `info` maximum
```suggestion
            logger.info("Defering: Cannot set PBM configurations, MongoDB has not yet started.")
```

Reply:
fixed, thank you